### ISSUE TITLE
[OSX] Reborn dies (semi-)silently on launch

### ISSUE THREAD

**Issue by mmdeas** [ValveSoftware/Dota-2#217]
When launching reborn, I get the "Preparing to launch Dota 2..." dialog box, my status changes to in-game and just over a second later changes back to online with no message.
Launching Steam from Terminal gives the following messages when attempting to launch Reborn:

```
Game update: AppID 570 "Dota 2", ProcID 1918, IP 0.0.0.0:0
FATAL: /Users/miguel/Library/Application Support/Steam/steamapps/common/dota 2 beta/game/bin/osx64/libengine2.dylib not loaded
Game removed: AppID 570 "Dota 2", ProcID 1918
```

Running OS X 10.8.5 on MacBookPro7,1.

EDIT: More info from Steam's System Information:

```
Processor Information:
    Model:  MacBookPro7,1
    Vendor:  GenuineIntel
    CPU Family:  0x6
    CPU Model:  0x17
    CPU Stepping:  0xa
    CPU Type:  0x0
    Speed:  2400 Mhz
    2 logical processors
    2 physical processors
    HyperThreading:  Unsupported
    FCMOV:  Supported
    SSE2:  Supported
    SSE3:  Supported
    SSSE3:  Supported
    SSE4a:  Unsupported
    SSE41:  Supported
    SSE42:  Unsupported

Network Information:
    Network Speed:  

Operating System Version:
    MacOS 10.8.5 (64 bit)

Video Card:
    Driver:  NVIDIA GeForce 320M

    Driver Version:  1.2
    Driver Date: 0  0
    OpenGL Version: 2.1
    Desktop Color Depth: 32 bits per pixel
    Monitor Refresh Rate: 0 Hz
    VendorID:  0x10de
    DeviceID:  0x8a0
    Number of Monitors:  1
    Number of Logical Video Cards:  1
    Primary Display Resolution:  1280 x 800
    Desktop Resolution: 1280 x 800
    Primary Display Size: 11.22" x 7.01"  (13.23" diag)
                                            28.5cm x 17.8cm  (33.6cm diag)
    Primary VRAM: 256 MB

Memory:
    RAM:  4096 Mb
```

**Comment by gdrewb-valve** [comment]
You don't get an OSX failure dialog?  Does the system information have any failure reports?  Has Reborn ever worked on the system?  Does Source 1 Dota 2 work?

**Comment by Najadorel** [comment]
I had a similar crash a while ago, did a "Verify Integrity of Gam Cache", downloaded a missing file and solve the issue, maybe its the same here

**Comment by mmdeas** [comment]
I get no dialog, it just goes back to where it was as if I'd closed the game.
The Steam System Information has no failure reports - the only things I didn't copy were the Miscellaneous and Installed Software sections.
I have never had Reborn working on this system.
Source 1 Dota 2 launches without problem (EDIT2: I haven't tried playing any games, only launching). After installing Reborn, Source 1 Dota 2 failed to launch (but with an error). Verifying the game cache fixed that. I have, however, uninstalled and reinstalled Reborn since then so I will attempt to verify the game cache again and report back.

EDIT: Verified game cache, no change.

**Comment by davidw-valve** [comment]
The Dota2 Reborn beta only supports OSX 10.9 and above. Is there a reason you cannot upgrade your version of OSX?

**Comment by mmdeas** [comment]
Ah. I did wonder but I couldn't find any information on the requirements for the beta.
I have some concerns about upgrading but they're beyond the scope of this issue.
Thanks for the information; at least I know it's not supposed to work now.

**Comment by davidw-valve** [comment]
No problem. I was just wondering the reason for you not upgrading since it would be easier for us to simply support recent versions of OSX, but if we have a substantial number of players who are on earlier versions of OSX and have good reasons for not upgrading we would love to support them!

**Comment by Tele42** [comment]
Apple bumped OpenGL support from 3.2 to 4.1 in OSX 10.9, which I assume is why source 2 is avoiding OSX 10.8 and older. Is this information (or just minimum supported OSX version) conveyed to the player at all?

**Comment by gdrewb-valve** [comment]
It's actually more about C++11 support, but we haven't set a minimum in stone yet.  We still have a ways to go on smoothing out problems in the code (in particular we have a huge range of perf on Macs) but once we've stabilized we'll have a better idea of where to make the cuts.

**Comment by mmdeas** [comment]
So basically, I heard that the upgrades seemed to use up more RAM and running low on memory is a problem I have had rather frequently already. This combined with the fact that I had to move my hard drive into my CD slot because it stopped working when attached to the original port meant that I didn't want to risk needing to downgrade while I actively needed my laptop to be working as I didn't know how easy it would be to get it back without a CD drive (this model only kinda boots from USB). Given how much of a pain it was to convince it to install Windows as a separate partition, I wasn't convinced it would be fun.
However, I've just graduated, am not working on anything particularly important and have plenty of time on my hands so maybe I'll take the time to try it out now and hope that I'm pleasantly surprised (and get to play Dota).
